lib/main.js

```diff
@@ -1,37 +1,45 @@
 'use strict';
 
-var autoprefixer = require('autoprefixer');
 var cheerio = require('cheerio');
+var autoprefixer = require('autoprefixer');
+
 var VERSION = require('../package.json').version;
 
-var HtmlAutoprefixer = function() {
+var HTMLAutoprefixer = function() {
   this.version = VERSION;
+  this.re = /#tmp\d+{(.+?)}/g;
   return this;
 };
 
-HtmlAutoprefixer.prototype.process = function(htmlString, cheerioOpts) {
+HTMLAutoprefixer.prototype.process = function(htmlString, cheerioOpts, autoprefixerOpts) {
   var $ = cheerio.load(htmlString, cheerioOpts);
   var styles = $('style').contents();
-  var elementsWithStyleTag = $('[style]');
+  var elementsWithStyleAttr = $('[style]');
   var prefixed;
 
   styles.each(function(index, style) {
-    var css = style.data;
-    prefixed = autoprefixer.process(css).css;
+    prefixed = autoprefixer.process(style.data, autoprefixerOpts).css;
     style.data = prefixed;
   });
 
-  elementsWithStyleTag.each(function(index, element) {
-    var css = '.tmp{' + element.attribs.style + '}';
-    try {
```

Comment:
@devinus this was mainly used to debug where it would fail throughout all the stylesheets :/

This + grep on the `style` made it easier to fix. I'm completely in favor of it being removed, but we will have to find a way to pinpoint the culprit when building templates.